app/controllers/Application.java

```diff
@@ -1953,10 +1956,85 @@ public static Result getUserAuthorizationStatus(String sessionId, String jobDefI
       logger.error("Error while fetching User's Project Authorization status ",ex);
       return internalServerError("Something went wrong while authorizing the user");
     }
-    logger.info(responseMap.toString());
     return ok(Json.toJson(responseMap));
   }
 
+  /** Rest API implementation for updating the TuneIn details like Tuning Param's value, AutoApply or #iterations
+   *
+   * @result JSON containing updated Tuning Params if params are changed and current value of AutoApply amd iterationCount
+   *
+   * {"updatedParamDetails":
+   *    {
+   *      "userSuggestedParamSetId":1501,
+   *      "tuningParameters":
+   *          [{
+   *            "paramId":21,
+   *           "name":"spark.driver.memory",
+   *           "currentParamValue":1512.04
+   *          },
+   *
+   *          {
+   *             "paramId":22,
+   *             "name":"spark.executor.cores",
+   *             "currentParamValue":90.0
+   *           },
+   *           {
+   *              "paramId":20,
+   *              "name":"spark.executor.memory",
+   *              "currentParamValue":7949.0
+   *            },
+   *            {
+   *               "paramId":23,
+   *               "name":"spark.memory.fraction",
+   *               "currentParamValue":0.14
+   *             }
+   *           ]
+   *    },
+   *    "autoApply":true,
+   *    "iterationCount":20
+   *  }
+   **/
+  public static Result updateTuneinDetails() {
```

Comment:
Where is this method called ?

Reply:
This is the controller method which is called when update Tunein info API is called. It is configured in the `routes` file.